source/agora/network/Client.d

```diff
@@ -223,7 +226,19 @@ public class NetworkClient
             break;
 
         case Preimage:
-            this.attemptRequest!(API.postPreimage, Throw.No)(event.preimage);
+            // Clean up the `last_preimages` priodically
+            // Note: It clears `last_preimages` everyday based on the CoinNet.
+            if (event.preimage.height % Height(144) == 0)
+                this.last_preimages.clear();
+
+            // We do not post the preimage if it is already posted in order to
+            // prevent calls of `postPreimage` from flooding
+            if (event.preimage.utxo !in this.last_preimages ||  
+                event.preimage.height > this.last_preimages[event.preimage.utxo].height)
```

Comment:
We only need to store the `Height` actually, not the full `PreImageInfo`, right ?